first-commit

slow6r opened the pull request.

slow6r committed:
first-commit

rnikolaenkov reviewed (commented):
Вынужден отклонить. Приложение не запускается. https://disk.yandex.ru/d/GmlzX8Dg8EVbbA
Необходимо привести зависимости в порядок (убрать неиспользуемые зависимсоти, добавить в зависимости библиотеки, которые используете) https://disk.yandex.ru/d/WeJ355fP99M2Bw

slow6r committed:
правки

rnikolaenkov commented on backend/src/controllers/auth.ts:
```diff
@@ -0,0 +1,67 @@
+import { validateOrderBody } from '../middlewares/validations'; // Импорт валидации
+import { OrderRequestBody } from '../types/interface';
+
+const createOrder = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
```
Вы, наверно, хотели тут организовать создание и авторизацию пользователя

rnikolaenkov reviewed (commented):
Отличная работа. **Но нужно исправить:**
1. Разобраться с зависимостями https://disk.yandex.ru/d/nJVsFZoI9R3Vvg
2. Убрать вывод информации в консоль (Вы используете логирование ошибок)
Обратите внимание, что Вы создали контроллер для работы с пользователями, но в нем метод для работы с заказами 

**Рекомендую для ознакомления:**
1. [8 правил, которые пригодятся при описании Git-коммитов](https://habr.com/ru/companies/sberbank/articles/662744/){target="_blank"}
2. [Соглашение о коммитах](https://www.conventionalcommits.org/ru/v1.0.0-beta.2/){target="_blank"}
3. [Основы Git – Хороший коммит, правила правильного коммита](https://truehub.ru/videos/git/good-commit){target="_blank"}
4. [Шпаргалки по безопасности: Nodejs](https://habr.com/ru/companies/acribia/articles/493714/){target="_blank"}
5. Рассказывается в контексте VueJS, но очень хорошо и кратко описаны основные принципы и шаблоны проектирования. [Принципы и шаблоны проектирования программного обеспечения](https://vue-faq.org/ru/book/Chapter_2__Software_Design_Principles_and_Patterns.html){target="_blank"}

slow6r committed:
chore: refactor config, improve error handling, and clean up dependencies

- Moved sensitive data (JWT secret) to config.ts
- Refactored auth middleware to use config for JWT secret
- Updated server initialization with improved logging
- Cleaned up unused dependencies and devDependencies (e.g., bcryptjs, joi)
- Improved error handling in app setup and database connection

rnikolaenkov commented on backend/src/controllers/auth.ts:
```diff
@@ -0,0 +1,90 @@
+
+    res.status(201).json(newUser);
+  } catch (err) {
+    if (err instanceof Error) {
```
**Можно лучше:** Старайтесь избегать вложенность условий. 

`
if (err instanceof Error && err.name === 'ValidationError') {
  next(new BadRequestError('Некорректные данные при создании пользователя'));
} else {
  next(err);
}
`

rnikolaenkov reviewed (approved):
Отличная работа над ошибками, все ошибки исправлены. Работа принята! В целом по работе: работа выполнена на высоком уровне, код хорошо структурирован. Названия переменных, классов и методов соответствуют общепринятым соглашениям (сразу понятно за что отвечают, что хранится и что выполняют).

**Рекомендую посмотреть:**

1.  [Шпаргалки по безопасности: Nodejs](https://habr.com/ru/companies/acribia/articles/493714/){target="_blank"}
2. [Использование переменных окружения](https://habr.com/ru/company/ruvds/blog/351254/){target="_blank"}
